File: backend/src/main/java/com/back/domain/member/service/AdminService.java
package com.back.domain.member.service;


import com.back.domain.member.dto.response.MemberResponseDto;
import com.back.domain.member.entity.Member;
import com.back.domain.member.exception.MemberErrorCode;
import com.back.domain.member.exception.MemberException;
import com.back.domain.member.repository.MemberRepository;
import com.back.global.exception.CustomException;
import com.back.global.exception.ErrorCode;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import java.util.List;
import java.util.stream.Collectors;

@Service
@Transactional(readOnly = true)
@RequiredArgsConstructor
public class AdminService {

    private final MemberRepository memberRepository;

    public List<MemberResponseDto> getAllMembers() {
        return memberRepository.findAll().stream()
                .map(MemberResponseDto::from)
                .collect(Collectors.toList());
    }

    public MemberResponseDto getMember(Long memberId) {
        Member member = memberRepository.findById(memberId)
                .orElseThrow(() -> new MemberException(MemberErrorCode.MEMBER_NOT_FOUND));
        return MemberResponseDto.from(member);
    }

    @Transactional
    public void deleteMember(Long memberId) {
        if (!memberRepository.existsById(memberId)) {
            throw new MemberException(MemberErrorCode.MEMBER_NOT_FOUND);
        }
        memberRepository.deleteById(memberId);
    }
}

File: backend/src/main/java/com/back/domain/member/service/AdminService.kt
package com.back.domain.member.service

import com.back.domain.member.dto.response.MemberResponseDto
import com.back.domain.member.entity.toDto
import com.back.domain.member.exception.MemberErrorCode
import com.back.domain.member.exception.MemberException
import com.back.domain.member.repository.MemberRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class AdminService(
    private val memberRepository: MemberRepository
) {

    fun getAllMembers(): List<MemberResponseDto> {
        return memberRepository.findAll().map { it.toDto() }
    }

    fun getMember(memberId: Long): MemberResponseDto {
        val member = memberRepository.findById(memberId)
            .orElseThrow { MemberException(MemberErrorCode.MEMBER_NOT_FOUND) }

        return member.toDto()
    }

    @Transactional
    fun deleteMember(memberId: Long) {
        if (!memberRepository.existsById(memberId)) {
            throw MemberException(MemberErrorCode.MEMBER_NOT_FOUND)
        }
        memberRepository.deleteById(memberId)
    }
}